lib/git_server/webserver.rb

```diff
@@ -66,5 +66,14 @@ class Webserver < Sinatra::Base
              content: content
            })
     end
+
+    get '/concepts/:track_slug/:concept_slug/data' do
+      data = RetrieveConceptData.(
+        params[:track_slug],
+        params[:concept_slug],
+        params[:git_sha] || "HEAD"
```

Comment:
I can't remember if I added `presence` to this. I think I did. If so, this would be better as it guards against empty strings.

```suggestion
        params[:git_sha].presence || "HEAD"
```

Reply:
You did. The tests still pass